xarray/backends/zarr.py

```diff
@@ -692,3 +700,233 @@ def maybe_chunk(name, var, chunks):
+            addition to the 'minimal' data variables.
+     coords : {'minimal', 'different', 'all' or list of str}, optional
+        These coordinate variables will be concatenated together:
+          * 'minimal': Only coordinates in which the dimension already appears
+            are included.
+          * 'different': Coordinates which are not equal (ignoring attributes)
+            across all datasets are also concatenated (as well as all for which
+            dimension already appears). Beware: this option may load the data
+            payload of coordinate variables into memory if they are not already
+            loaded.
+          * 'all': All coordinate variables will be concatenated, except
+            those corresponding to other dimensions.
+          * list of str: The listed coordinate variables will be concatenated,
+            in addition the 'minimal' coordinates.
+        parallel : bool, optional
+            If True, the open and preprocess steps of this function will be
+            performed in parallel using ``dask.delayed``. Default is False.
+    join : {'outer', 'inner', 'left', 'right', 'exact, 'override'}, optional
+        String indicating how to combine differing indexes
+        (excluding concat_dim) in objects
+        - 'outer': use the union of object indexes
+        - 'inner': use the intersection of object indexes
+        - 'left': use indexes from the first object with each dimension
+        - 'right': use indexes from the last object with each dimension
+        - 'exact': instead of aligning, raise `ValueError` when indexes to be
+          aligned are not equal
+        - 'override': if indexes are of same size, rewrite indexes to be
+          those of the first object with that dimension. Indexes for the same
+          dimension must have the same size in all objects.
+    attrs_file : str or pathlib.Path, optional
+        Path of the file used to read global attributes from.
+        By default global attributes are read from the first file provided,
+        with wildcard matches sorted by filename.
+    **kwargs : optional
+        Additional arguments passed on to :py:func:`xarray.open_zarr`.
+
+
+    Returns
+    -------
+    xarray.Dataset
+    Notes
+    -----
+    ``open_zarr`` opens files with read-only access. When you modify
+    values
+    of a Dataset, even one linked to files on disk, only the in-memory copy you
+    are manipulating in xarray is modified: the original file on disk is never
+    touched.
+    See Also
+    --------
+    combine_by_coords
+    combine_nested
+    open_dataset
+    References
+    ----------
+    .. [1] http://xarray.pydata.org/en/stable/dask.html
+    .. [2] http://xarray.pydata.org/en/stable/dask.html#chunking-and-performance
+    """
+    if isinstance(paths, str):
+        if is_remote_uri(paths):
+            raise ValueError(
+                "cannot do wild-card matching for paths that are remote URLs: "
```

Comment:
In theory, you should be able to do this. The trouble is matching directories? It may be worthwhile writing out how this ought to work and making a test case for fsspec. 

The code here could maybe also be gathered into some "open_mappers" function in fsspec, which parses the glob and opens the mappers. I'm not sure how useful that idea is, maybe its better to keep the code here, likely the only place it will get used.

Reply:
An idea is to have (on ```intake-xarray```) a conditional that **if** paths is glob, ```intake-xarray``` passes the glob path as is to ```open_mzarr``` (i.e. not use ```open_mapper```). As written here, ```xarray.open_mzarr``` can sort the glob and open the zarr stores. However, we could make use of ```open_mappers``` corresponding to entries on the sorted ```paths``` list (line 851), to open the zarr stores. Potentially modifying how ```open_kwargs`` (line 870) and ```datasets``` (886) are defined in order to make use of ```fsspec.open_mappers``` .

A question that comes to my mind (as I am new at this), is what is the advantage of employing ```open_mapper``` to open a zarr store? That is, as opposed to provide the location of the store to ```open_zarr``` the way it is currently defined on ```datasets``` (line 886), which does work... performance?